ownCloud/Client/Sharing/GroupSharingTableViewController.swift

```diff
@@ -0,0 +1,473 @@
+
+	// MARK: - Init & Deinit
+
+	public init(core inCore: OCCore, item inItem: OCItem) {
+		core = inCore
+		item = inItem
+
+		if item.isShareable {
+			if item.isSharedWithUser == false {
+				meCanShareItem = true
+			} else if item.isSharedWithUser, core?.connection.capabilities?.sharingResharing == true {
+				meCanShareItem = true
+			}
+		}
+
+		super.init(style: .grouped)
+	}
+
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		if meCanShareItem {
+			searchController = UISearchController(searchResultsController: nil)
+			searchController?.searchResultsUpdater = self
+			searchController?.hidesNavigationBarDuringPresentation = true
+			searchController?.dimsBackgroundDuringPresentation = false
+			searchController?.searchBar.placeholder = "Search User, Group, Remote".localized
+			searchController?.searchBar.delegate = self
+			navigationItem.hidesSearchBarWhenScrolling = false
+			navigationItem.searchController = searchController
+			definesPresentationContext = true
+			searchController?.searchBar.applyThemeCollection(Theme.shared.activeCollection)
+			recipientSearchController = core?.recipientSearchController(for: item)
+			recipientSearchController?.delegate = self
+		}
+
+		messageView = MessageView(add: self.view)
+
+		self.navigationItem.title = "Sharing".localized
+		self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(dismissView))
+
+		addHeaderView()
+
+		shareQuery = core?.sharesWithReshares(for: item, initialPopulationHandler: { (sharesWithReshares) in
+			if sharesWithReshares.count > 0 {
+				self.shares = sharesWithReshares.filter { (OCShare) -> Bool in
+					if OCShare.type != .link {
+						return true
+					}
+					return false
+				}
+				OnMainThread {
+					self.addShareSections()
+				}
+			}
+			_ = self.core?.sharesSharedWithMe(for: self.item, initialPopulationHandler: { (sharesWithMe) in
+				OnMainThread {
+					if sharesWithMe.count > 0 {
+						var shares : [OCShare] = []
+						shares.append(contentsOf: sharesWithMe)
+						shares.append(contentsOf: sharesWithReshares)
+						self.shares = shares
+						self.removeShareSections()
+						self.addShareSections()
+					}
+
+					self.addActionShareSection()
+				}
+			})
+
+
+		}, changesAvailableNotificationHandler: { (sharesWithReshares) in
+			let sharesWithReshares = sharesWithReshares.filter { (OCShare) -> Bool in
+				if OCShare.type != .link {
+					return true
+				}
+				return false
+			}
+			self.shares = sharesWithReshares
+			OnMainThread {
+				self.removeShareSections()
+				self.addShareSections()
+
+				self.addActionShareSection()
+			}
+		})
+		shareQuery?.refreshInterval = 2
+	}
+
+	@objc func dismissView() {
+		if let query = self.shareQuery {
+			self.core?.stop(query)
+		}
+		dismissAnimated()
+	}
+
+	// MARK: - Header View
+
+	func addHeaderView() {
```

Comment:
This code is shared between `GroupSharingTableViewController` and `PublicLinkTableViewController`. Maybe create a parent class to move this code to, so it can be reused in both.